src/app/content/components/SearchResultsSidebar/SearchResultsBarWrapper.tsx

```diff
@@ -0,0 +1,83 @@
+import { SearchResultContainer } from '../../search/types';
+import { Book } from '../../types';
+import * as Styled from './styled';
+
+interface SearchResultsSidebarProps {
+  query: string | null;
+  totalHits: number | null;
+  results: SearchResultContainer[] | null;
+  onClose: () => void;
+  toggleSearchSidebar: (open: boolean) => void;
+  open: boolean;
+  book?: Book;
+}
+
+export class SearchResultsBarWrapper extends Component<SearchResultsSidebarProps> {
+  public searchSidebar = React.createRef<HTMLElement>();
+
+  public render() {
+    const {query, totalHits, results, onClose, book, open, toggleSearchSidebar} = this.props;
+
+    return <Styled.SearchResultsBar ref={this.searchSidebar} open={open}>
+      {!results && <Styled.LoadingWrapper>
+        <Styled.CloseIconWrapper>
+          <Styled.CloseIconButton onClick={onClose}><Styled.CloseIcon /></Styled.CloseIconButton>
+        </Styled.CloseIconWrapper>
+        <Loader/>
+      </Styled.LoadingWrapper>}
+      {totalHits && <Styled.SearchQueryWrapper>
+        <FormattedMessage id='i18n:search-results:bar:query:results'>
+          {(msg: Element | string) =>
+            <Styled.SearchQuery>
+              <Styled.SearchIconInsideBar />
+              <Styled.HeaderQuery>
+                {totalHits} {msg} <strong> &lsquo;{query}&rsquo;</strong>
+              </Styled.HeaderQuery>
+            </Styled.SearchQuery>
+          }
+        </FormattedMessage>
+        <Styled.CloseIconButton onClick={onClose}><Styled.CloseIcon /></Styled.CloseIconButton>
+      </Styled.SearchQueryWrapper>}
+      {!totalHits && <div>
+        <Styled.CloseIconWrapper>
+          <Styled.CloseIconButton onClick={onClose}><Styled.CloseIcon /></Styled.CloseIconButton>
+        </Styled.CloseIconWrapper>
+        <FormattedMessage id='i18n:search-results:bar:query:no-results'>
+          {(msg: Element | string) =>
+            <Styled.SearchQuery>
+              <Styled.SearchQueryAlignment>{msg} <strong> &lsquo;{query}&rsquo;</strong></Styled.SearchQueryAlignment>
+            </Styled.SearchQuery>
+          }
+        </FormattedMessage>
+      </div>}
+      {book && results && totalHits && <Styled.NavOl>
+        <SearchResultContainers containers={results} book={book} toggle={toggleSearchSidebar}/>
+      </Styled.NavOl>}
+      </Styled.SearchResultsBar>;
+  }
+
+  public componentDidMount = () => {
+    const searchSidebar = this.searchSidebar.current;
+
+    if (!searchSidebar || typeof(window) === 'undefined') {
+      return;
+    }
+
+    const scrollHandler = () => {
+      const top = searchSidebar.getBoundingClientRect().top;
+      searchSidebar.style.setProperty('height', `calc(100vh - ${top}px)`);
+    };
+
+    const animation = () => requestAnimationFrame(scrollHandler);
+
+    window.addEventListener('scroll', animation, {passive: true});
+    window.addEventListener('resize', animation, {passive: true});
+    scrollHandler();
```

Comment:
try to move this logic into a util function to be shared with the `scrollHandler` in the ToC